src/main/java/com/iexec/sms/secret/web3/Web3Secret.java

```diff
@@ -17,29 +17,54 @@
 package com.iexec.sms.secret.web3;
 
 import com.iexec.sms.secret.Secret;
-import lombok.Data;
-import lombok.EqualsAndHashCode;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
+import lombok.*;
 import org.hibernate.annotations.GenericGenerator;
 
 import javax.persistence.Entity;
 import javax.persistence.GeneratedValue;
 import javax.persistence.Id;
 
-@EqualsAndHashCode(callSuper = true)
-@Data
-@Getter
 @Entity
-@NoArgsConstructor
+@Getter
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
 public class Web3Secret extends Secret {
 
     @Id
     @GeneratedValue(generator = "system-uuid")
     @GenericGenerator(name = "system-uuid", strategy = "uuid")
     private String id;
 
-    public Web3Secret(String address, String value) {
-        super(address, value);
+    private Web3Secret(String superId, String id, String address, String value, boolean isEncryptedValue) {
```

Comment:
Why do we introduce `superId`?

Reply:
We don't introduce anything. `Secret` has a `id` field, so does `Web3Secret`. This means `Web3Secret` has one `id` and one `super.id` field. We have to copy it when copying a `Web3Secret` object.

Reply:
Can you please add a `TODO` about cleaning multiple IDs so we don't miss it later :) ?

Reply:
Thank you :pray: